db/async_wait.c

```diff
@@ -0,0 +1,869 @@
+                if (gbl_async_dist_commit_verbose) {
+                    logmsg(LOGMSG_USER, "doing slow node check at time %d\n", now);
+                }
+                bdb_slow_replicant_check(item->bdb_state, &item->seqnum);
+            }
+        }
+        item->wait_init_end_time = comdb2_time_epochms();
+    case WAIT_FOR_FIRST_ACK:
+        if (item->cur_state != WAIT_FOR_FIRST_ACK) {
+            item->cur_state = WAIT_FOR_FIRST_ACK;
+            item->wait_wait_for_first_ack_start_time = comdb2_time_epochms();
+        }
+        if (((comdb2_time_epochms() - item->start_time) < item->bdb_state->attr->rep_timeout_maxms) &&
+            !(lock_desired = bdb_lock_desired(item->bdb_state))) {
+            item->numnodes = 0;
+            item->numskip = 0;
+            item->numwait = 0;
+            item->num_commissioned = net_get_all_commissioned_nodes(item->bdb_state->repinfo->netinfo, item->connlist);
+
+            if (item->num_commissioned == 0) {
+                /* no replicants to wait for! */
+                if (gbl_async_dist_commit_verbose) {
+                    logmsg(LOGMSG_USER, "found %d nodes commissioned. Done waiting!\n", item->num_commissioned);
+                }
+                item->cur_state = WAIT_FINISH;
+                item->wait_wait_for_first_ack_end_time = comdb2_time_epochms();
+                goto wait_finish_label;
+            }
+            for (int i = 0; i < item->num_commissioned; i++) {
+                int wait = 0;
+                /* is_incoherent returns 0 for COHERENT & INCOHERENT_WAIT */
+                if (!(is_incoherent_complete(item->bdb_state, item->connlist[i], &wait))) {
+                    item->nodelist[item->numnodes] = item->connlist[i];
+                    item->numnodes++;
+                    if (wait)
+                        item->numwait++;
+                } else {
+                    item->numskip++;
+                    item->num_incoh++;
+                }
+            }
+
+            if (item->numnodes == 0) {
+                item->cur_state = WAIT_FINISH;
+                item->wait_wait_for_first_ack_end_time = comdb2_time_epochms();
+                goto wait_finish_label;
+            }
+            if (gbl_async_dist_commit_verbose) {
+                logmsg(LOGMSG_USER, "item->numnodes: %d, item->numskip: %d, item->numwait: %d, item->num_incoh: %d\n",
+                       item->numnodes, item->numskip, item->numwait, item->num_incoh);
+            }
+            for (int i = 0; i < item->numnodes; i++) {
+                if (item->bdb_state->rep_trace) {
+                    logmsg(LOGMSG_USER, "waiting for initial NEWSEQ from node %s of >= <%d:%d>\n", item->nodelist[i],
+                           item->seqnum.lsn.file, item->seqnum.lsn.offset);
+                }
+                int rc = bdb_wait_for_seqnum_from_node_int(item->bdb_state, &item->seqnum, item->nodelist[i], 1000,
```

Comment:
Discussing this actively: the api will do a cond-wait inline for 1000 ms .. I think it is wrong here ..